presto-main/src/test/java/io/prestosql/operator/BenchmarkUnnestOperator.java

```diff
@@ -0,0 +1,424 @@
+        {
+            if (typeString.equals("NONE")) {
+                return Optional.empty();
+            }
+            TypeSignature signature = TypeSignature.parseTypeSignature(typeString);
+            return Optional.of(typeManager.getType(signature));
+        }
+
+        @TearDown
+        public void cleanup()
+        {
+            executor.shutdownNow();
+            scheduledExecutor.shutdownNow();
+        }
+
+        public TaskContext createTaskContext()
+        {
+            return TestingTaskContext.createTaskContext(executor, scheduledExecutor, TEST_SESSION, new DataSize(2, GIGABYTE));
+        }
+
+        public OperatorFactory getOperatorFactory()
+        {
+            return operatorFactory;
+        }
+
+        public List<Page> getPages()
+        {
+            return pages;
+        }
+
+        private static List<Page> createInputPages(int positionsPerPage, List<Type> types, InputGenerator inputGenerator, double primitiveNullsUnnest, double rowNullsUnnest)
+        {
+            ImmutableList.Builder<Page> pages = ImmutableList.builder();
+            int numPages = TOTAL_POSITIONS / positionsPerPage;
+
+            for (int i = 0; i < numPages; i++) {
+                Block[] blocks = new Block[types.size()];
+                blocks[0] = inputGenerator.produceBlock(types.get(0), positionsPerPage, 0.0, 0.0);
+                blocks[1] = inputGenerator.produceBlock(types.get(1), positionsPerPage, primitiveNullsUnnest, rowNullsUnnest);
+                if (blocks.length == 3) {
+                    blocks[2] = inputGenerator.produceBlock(types.get(2), positionsPerPage, 0.0, 0.0);
+                }
+                pages.add(new Page(blocks));
+            }
+
+            return pages.build();
+        }
+    }
+
+    @Benchmark
+    public List<Page> unnest(BenchmarkContext context)
+    {
+        DriverContext driverContext = context.createTaskContext().addPipelineContext(0, true, true, false).addDriverContext();
+        Operator operator = context.getOperatorFactory().createOperator(driverContext);
+
+        Iterator<Page> input = context.getPages().iterator();
+        ImmutableList.Builder<Page> outputPages = ImmutableList.builder();
+
+        boolean finishing = false;
+        for (int loops = 0; !operator.isFinished() && loops < 1_000_000; loops++) {
+            if (operator.needsInput()) {
+                if (input.hasNext()) {
+                    Page inputPage = input.next();
+                    operator.addInput(inputPage);
+                }
+                else if (!finishing) {
+                    operator.finish();
+                    finishing = true;
+                }
+            }
+
+            Page outputPage = operator.getOutput();
+            if (outputPage != null) {
+                outputPages.add(outputPage);
+            }
+        }
+
+        return outputPages.build();
+    }
+
+    /**
+     * Helper class for random input generation.
```

Comment:
Remove this comment. It doesn't really add any info that's not obvious from the code below.